src/extension/intellisenseHelper.ts

```diff
@@ -53,21 +69,53 @@ export class IntellisenseHelper {
     }
 
     /**
-     * Helper method that installs Typescript into a global location.
+     * Helper method that verifies the correct version of TypeScript is installed.
+     * If using a newer version of VSCode TypeScript is installed by default and no
+     * action is needed. If using an older version, verify that the correct TS version is
+     * installed, if not install it.
      */
-    public static installTypescriptNext(isRestartRequired: boolean): Q.Promise<boolean> {
-        let typeScriptNextDest: string = path.resolve(IntellisenseHelper.getUserHomePath(), ".vscode");
-        let typeScriptNextLibPath: string = path.join(typeScriptNextDest, "node_modules", "typescript", "lib");
-        let fileSystem: FileSystem = new FileSystem();
+    public static verifyInstallTypeScript(isRestartRequired: boolean): Q.Promise<boolean> {
 
-        return fileSystem.exists(typeScriptNextLibPath)
-            .then(function(exists: boolean) {
-                if (!exists) {
-                    return Q.nfcall(child_process.exec, `npm install --prefix ${typeScriptNextDest} typescript@next`)
-                        .then(() => { return true; });
+        if (IntellisenseHelper.isSalsaSupported()) {
+            // this is the correct version of vscode, which includes TypeScript (Salsa) support, nothing to do here
+            return Q.resolve<boolean>(isRestartRequired);
+        }
+
+        return IntellisenseHelper.getInstalledTypeScriptVersion()
+            .then(function(installProps: InstallProps) {
+
+                if (installProps.installed === false) {
+
+                    if(semver.neq(IntellisenseHelper.s_typeScriptVersion, installProps.version)) {
+                        Log.logInternalMessage(LogLevel.Debug, "TypeScript is installed with the wrong version: " + installProps.version);
+                    }
+                    else {
+                        Log.logInternalMessage(LogLevel.Debug, "TypeScript is not installed");
+                    }
+                    return Q.resolve<boolean>(true);
                 }
 
-                return isRestartRequired;
+                Log.logInternalMessage(LogLevel.Debug, "Installed TypeScript version is correct");
+                return Q.resolve<boolean>(false);
+            })
+            .then( (install: boolean) => {
+
+                if(install) {
+                    let installPath: string = path.resolve(IntellisenseHelper.getUserHomePath(), ".vscode");
+                    let installParam: string = "npm install --prefix " + installPath + " typescript@" + IntellisenseHelper.s_typeScriptVersion;
+                    Log.logInternalMessage(LogLevel.Debug, "Installing TypeScript: " + installParam);
+                    return Q.nfcall(child_process.exec, installParam)
```

Comment:
We've tried to avoid directly calling child_process in favor of using our own common/node/childProcess wrapper that already promisifies things. It should make things easier for testing in future since we know that all side effects (running processes / messing with filesystem) go through just a couple of locations.